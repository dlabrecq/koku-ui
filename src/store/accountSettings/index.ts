import * as accountSettingsActions from './accountSettingsActions';
import { stateKey as accountSettingsStateKey } from './accountSettingsCommon';
import { AccountSettingsAction, accountSettingsReducer, AccountSettingsState } from './accountSettingsReducer';
import * as accountSettingsSelectors from './accountSettingsSelectors';

export {
  AccountSettingsAction,
  accountSettingsActions,
  accountSettingsReducer,
  accountSettingsSelectors,
  AccountSettingsState,
  accountSettingsStateKey,
};
