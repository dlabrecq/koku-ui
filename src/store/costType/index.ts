import * as costTypeActions from './costTypeActions';
import { stateKey as costTypeStateKey } from './costTypeCommon';
import { CostTypeAction, costTypeReducer, CostTypeState } from './costTypeReducer';
import * as costTypeSelectors from './costTypeSelectors';

export { CostTypeAction, costTypeActions, costTypeReducer, costTypeSelectors, CostTypeState, costTypeStateKey };
