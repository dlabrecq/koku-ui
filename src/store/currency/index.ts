import * as currencyActions from './currencyActions';
import { stateKey as currencyStateKey } from './currencyCommon';
import { CurrencyAction, currencyReducer, CurrencyState } from './currencyReducer';
import * as currencySelectors from './currencySelectors';

export { CurrencyAction, currencyActions, currencyReducer, currencySelectors, CurrencyState, currencyStateKey };
