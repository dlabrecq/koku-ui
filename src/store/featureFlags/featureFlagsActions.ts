import { createAction } from 'typesafe-actions';

export interface FeatureFlagsActionMeta {
  isCostCategoriesFeatureEnabled?: boolean;
  isCostDistributionFeatureEnabled?: boolean;
  isExportsFeatureEnabled?: boolean;
  isFinsightsFeatureEnabled?: boolean;
  isIbmFeatureEnabled?: boolean;
  isRosFeatureEnabled?: boolean;
  isSettingsFeatureEnabled?: boolean;
  isSettingsPlatformFeatureEnabled?: boolean;
}

export const setFeatureFlags = createAction('feature/init_feature_flags')<FeatureFlagsActionMeta>();
export const resetState = createAction('feature/reset_state')();
