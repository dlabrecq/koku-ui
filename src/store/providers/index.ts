import {
  awsProvidersQuery,
  azureProvidersQuery,
  gcpProvidersQuery,
  ibmProvidersQuery,
  ocpProvidersQuery,
  providersQuery,
} from 'store/providers/providersCommon';

import * as providersActions from './providersActions';
import { stateKey as providersStateKey } from './providersCommon';
import { ProvidersAction, providersReducer, ProvidersState } from './providersReducer';
import * as providersSelectors from './providersSelectors';

export {
  awsProvidersQuery,
  azureProvidersQuery,
  gcpProvidersQuery,
  ibmProvidersQuery,
  ocpProvidersQuery,
  ProvidersAction,
  providersActions,
  providersQuery,
  providersReducer,
  providersSelectors,
  ProvidersState,
  providersStateKey,
};
