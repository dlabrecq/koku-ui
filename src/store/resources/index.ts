import * as resourceActions from './resourceActions';
import { resourceStateKey } from './resourceCommon';
import { CachedResource, ResourceAction, resourceReducer, ResourceState } from './resourceReducer';
import * as resourceSelectors from './resourceSelectors';

export {
  ResourceAction,
  CachedResource,
  resourceActions,
  resourceReducer,
  resourceSelectors,
  ResourceState,
  resourceStateKey,
};
