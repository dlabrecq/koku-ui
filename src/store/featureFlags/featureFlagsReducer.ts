import { ActionType, getType } from 'typesafe-actions';

import { resetState, setFeatureFlags } from './featureFlagsActions';

export type FeatureFlagsAction = ActionType<typeof setFeatureFlags | typeof resetState>;

export type FeatureFlagsState = Readonly<{
  isCurrencyFeatureEnabled: boolean;
  isExportsFeatureEnabled: boolean;
  isIbmFeatureEnabled: boolean;
  isOciFeatureEnabled: boolean;
}>;

export const defaultState: FeatureFlagsState = {
  isCurrencyFeatureEnabled: false,
  isExportsFeatureEnabled: false,
  isIbmFeatureEnabled: false,
  isOciFeatureEnabled: false,
};

export const stateKey = 'featureFlags';

export function featureFlagsReducer(state = defaultState, action: FeatureFlagsAction): FeatureFlagsState {
  switch (action.type) {
    case getType(resetState):
      state = defaultState;
      return state;

    case getType(setFeatureFlags):
      return {
        ...state,
        isCurrencyFeatureEnabled: action.payload.isCurrencyFeatureEnabled,
        isExportsFeatureEnabled: action.payload.isExportsFeatureEnabled,
        isIbmFeatureEnabled: action.payload.isIbmFeatureEnabled,
        isOciFeatureEnabled: action.payload.isOciFeatureEnabled,
      };

    default:
      return state;
  }
}
