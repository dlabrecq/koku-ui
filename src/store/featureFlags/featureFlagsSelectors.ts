import { RootState } from 'store/rootReducer';

import { stateKey } from './featureFlagsReducer';

export const selectFeatureFlagsState = (state: RootState) => state[stateKey];

export const selectIsCurrencyFeatureEnabled = (state: RootState) =>
  selectFeatureFlagsState(state).isCurrencyFeatureEnabled;
export const selectIsExportsFeatureEnabled = (state: RootState) =>
  selectFeatureFlagsState(state).isExportsFeatureEnabled;
export const selectIsIbmFeatureEnabled = (state: RootState) => selectFeatureFlagsState(state).isIbmFeatureEnabled;
export const selectIsOciFeatureEnabled = (state: RootState) => selectFeatureFlagsState(state).isOciFeatureEnabled;
