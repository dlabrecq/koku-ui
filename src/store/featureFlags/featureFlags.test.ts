import { createMockStoreCreator } from 'store/mockStore';

import { featureFlagsSelectors } from '.';
import * as actions from './featureFlagsActions';
import { featureFlagsReducer, stateKey } from './featureFlagsReducer';
import * as selectors from './featureFlagsSelectors';

const createUIStore = createMockStoreCreator({
  [stateKey]: featureFlagsReducer,
});

test('default state', async () => {
  const store = createUIStore();
  expect(selectors.selectFeatureFlagsState(store.getState())).toMatchSnapshot();
});

test('currency feature is enabled', async () => {
  const store = createUIStore();
  store.dispatch(actions.setFeatureFlags({ isCurrencyFeatureEnabled: true }));
  expect(featureFlagsSelectors.selectIsCurrencyFeatureEnabled(store.getState())).toBe(true);
});

test('exports feature is enabled', async () => {
  const store = createUIStore();
  store.dispatch(actions.setFeatureFlags({ isExportsFeatureEnabled: true }));
  expect(featureFlagsSelectors.selectIsExportsFeatureEnabled(store.getState())).toBe(true);
});

test('IBM feature is enabled', async () => {
  const store = createUIStore();
  store.dispatch(actions.setFeatureFlags({ isIbmFeatureEnabled: true }));
  expect(featureFlagsSelectors.selectIsIbmFeatureEnabled(store.getState())).toBe(true);
});

test('OCI feature is enabled', async () => {
  const store = createUIStore();
  store.dispatch(actions.setFeatureFlags({ isOciFeatureEnabled: true }));
  expect(featureFlagsSelectors.selectIsOciFeatureEnabled(store.getState())).toBe(true);
});
