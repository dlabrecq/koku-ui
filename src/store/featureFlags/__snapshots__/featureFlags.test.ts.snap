// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default state 1`] = `
Object {
  "isCurrencyFeatureEnabled": false,
  "isExportsFeatureEnabled": false,
  "isIbmFeatureEnabled": false,
  "isOciFeatureEnabled": false,
}
`;
